src/leap/keymanager/openpgp.py

```diff
@@ -447,23 +434,30 @@ def _put_key_doc(self, key):
         def check_and_put(docs, key):
             if len(docs) == 1:
                 doc = docs.pop()
-                if key.fingerprint == doc.content[KEY_FINGERPRINT_KEY]:
+                oldkey = build_key_from_dict(OpenPGPKey, doc.content)
+                if key.fingerprint == oldkey.fingerprint:
                     # in case of an update of the key merge them with gnupg
                     with self._temporary_gpgwrapper() as gpg:
-                        gpg.import_keys(doc.content[KEY_DATA_KEY])
+                        gpg.import_keys(oldkey.key_data)
                         gpg.import_keys(key.key_data)
                         gpgkey = gpg.list_keys(secret=key.private).pop()
-                        key = _build_key_from_gpg(
+                        mergedkey = self._build_key_from_gpg(
                             gpgkey,
                             gpg.export_keys(gpgkey['fingerprint'],
                                             secret=key.private))
-                    doc.set_json(key.get_json())
+                    mergedkey.validation = max(
+                        [key.validation, oldkey.validation])
+                    mergedkey.last_audited_at = oldkey.last_audited_at
+                    mergedkey.refreshed_at = key.refreshed_at
+                    mergedkey.encr_used = key.encr_used or oldkey.encr_used
```

Comment:
If the new key was never used for encryption or signed, why should it have the same status as the new key?